File: backend/src/controllers/professionalController.js
import { ProfessionalService } from '../services/professionalService.js';

const service = new ProfessionalService();

export async function postProfissional(req, res, next) {
  try {
    const out = await service.cadastrar(req.body || {});
    res.status(201).json(out);
  } catch (e) {
    next(e);
  }
}

export async function getProfissionais(req, res, next) {
  try {
    const list = await service.listar();
    res.json(list);
  } catch (e) {
    next(e);
  }
}

export async function patchProfissionalAtivo(req, res, next) {
  try { res.json(await service.atualizarAtivo(req.params.id, req.body?.ativo)); } catch (e) { next(e); }
}

export async function deleteProfissional(req, res, next) {
  try { res.json(await service.excluir(req.params.id)); } catch (e) { next(e); }
}

export async function getServicos(req, res, next) {
  try {
    const list = await service.listarServicos();
    res.json(list);
  } catch (e) {
    next(e);
  }
}

File: backend-java/src/main/java/com/clinica/bot/controller/ProfessionalController.java
package com.clinica.bot.controller;

import com.clinica.bot.domain.Profissional;
import com.clinica.bot.security.SecurityUtils;
import com.clinica.bot.service.ProfessionalService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequiredArgsConstructor
public class ProfessionalController {

    private final ProfessionalService service;

    @PostMapping("/profissionais")
    @ResponseStatus(HttpStatus.CREATED)
    public Profissional create(@RequestBody Map<String, Object> body) {
        SecurityUtils.requireUser();
        return service.cadastrar(body);
    }

    @GetMapping("/profissionais")
    public List<Profissional> list() {
        return service.listar();
    }

    @PatchMapping("/profissionais/{id}/ativo")
    public Profissional patchAtivo(@PathVariable String id, @RequestBody Map<String, Object> body) {
        SecurityUtils.requireUser();
        return service.atualizarAtivo(id, body.get("ativo") instanceof Boolean b ? b : null);
    }

    @DeleteMapping("/profissionais/{id}")
    public Map<String, Object> delete(@PathVariable String id) {
        SecurityUtils.requireUser();
        return service.excluir(id);
    }

    @GetMapping("/servicos")
    public List<String> servicos() {
        return service.listarServicos();
    }
}
